src/core/document.js

```diff
@@ -821,76 +821,30 @@ class PDFDocument {
     });
   }
 
-  get xfaDatasets() {
+  get xfaStreams() {
```

Comment:
I don't think that we want to use `shadow` in this helper method, since that'll mean caching of Streams and those can be arbitrarily large once decoded (in the general case). Note how we, for that very reason, are purposely *not caching* Streams on the `XRef`-instance.

Also, maybe name it `_xfaStreams` instead to signal that it shouldn't be used directly?

Reply:
Yes and it's especially useless because we don't have to call this function so many times.